sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -157,10 +157,18 @@ object MultiFileReaderThreadPool extends Logging {
 
   /**
    * Get the existing thread pool or create one with the given thread count if it does not exist.
-   * @note The thread number will be ignored if the thread pool is already created.
+   * @note The thread number will be ignored if the thread pool is already created, or modified
+   *       if it is not the right size compared to the number of cores available.
    */
-  def getOrCreateThreadPool(numThreads: Int): ThreadPoolExecutor = {
-    threadPool.getOrElse(initThreadPool(numThreads))
+  def getOrCreateThreadPool(numThreadsFromConf: Int): ThreadPoolExecutor = {
+    threadPool.getOrElse {
+      val numThreads = Math.max(numThreadsFromConf, GpuDeviceManager.getNumCores)
+
+      if (numThreadsFromConf != numThreads) {
+        logWarning(s"Using $numThreads as the number of threads for the thread pool.")
```

Comment:
@gerashegalov 

> This setting is orthogonal to cores

The reason for this originally was to avoid bad performance if a user set the thread pool size too small. The idea was that we should not have a thread pool that is smaller than the number of tasks, or else we would lose parallelism by enabling the multi-threaded reader. In this case the number of cores is being used as a stand-in for tasks, even though it is not necessarily a one to one mapping.

> what do we expect the users to do upon seeing this message in the log?

I don't expect a user to do much of anything with this message right now. It is in the executor log and it is unlikely that a user will ever see it. It is more here so we have something that is equivalent to the log message on the driver, which is the one most likely to catch this. I also thought that it would be nice to know that this is happening, but you are right it is unlikely to ever be acted on. If you want me to remove it I will.

Reply:
thanks for elaborating @revans2.

> It is in the executor log and it is unlikely that a user will ever see it.

I would not count on it: Devops  look at logs by querying log search engines using app id using the likes of Splunk, Sumo, and Kibana rather than raw  individual logs. And they could track metrics for different log categories. 

> I also thought that it would be nice to know that this is happening, but you are right it is unlikely to ever be acted on. If you want me to remove it I will.

I think this message can be more useful if you add details to make it more self-contained.

"Configuring the file reader thread pool size with the max of ${CONF1}=$VAL1 and  ${CONF2}=${VAL2}"